clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -180,6 +186,25 @@ public class AsyncKafkaConsumer<K, V> implements ConsumerDelegate<K, V> {
      */
     private class BackgroundEventProcessor implements EventProcessor<BackgroundEvent> {
 
+        private Optional<StreamsRebalanceListener> streamsGroupRebalanceCallbacks = Optional.empty();
```

Comment:
nit: the field is named `streamsGroupRebalanceCallbacks` but the type wrapped by the `Optional` is `StreamsRebalanceListener` maybe rename to `streamsRebalanceListener`? Up to you